### DISCUSSION TITLE
Interpreter status

### DISCUSSION THREAD

**Discussion by notramo** [luckyframework/lucky#1849] [Q&A]
I know that the interpreter is still experimental, but I'm interested that exactly what are the stoppers in using `crystal i` for Lucky development?

**Comment by jwoertink** [comment]
I think there's still an issue that causes some problems https://github.com/crystal-lang/crystal/issues/12241#issuecomment-1181297104

But it sounds like for the most part it may be usable. I've never tried the interpreter though. If you do give it a try, and see any issues that we can fix on our end, feel free to open issues on them, or PRs.

**Reply by notramo** [reply]
I tried importing `src/app.cr` into the interpreter, but it says file not found.
I use the `84codes/crystal:1.10.1-ubuntu-22.04` Docker image as my computer don't have enough resources to compile a custom built Crystal compiler. I mounted the Lucky app to `/app`, and `Dir.current` and `Dir.children "."` gives correct results in the interpreter, but `require "./src/app.cr"` shows error.

**Comment by notramo** [comment]
I did some tests, it can answer requests that don't involve database queries, but it fails with Postgres.
However, while some Crystal apps start faster in the interpreter, the Lucky server start more than 2x slower. So until the interpreter compilation performance is improved for this usecase, it's better to compile.
